src/lib/SX127xDriver/SX127x.cpp

```diff
@@ -348,46 +357,49 @@ void ICACHE_RAM_ATTR SX127xDriver::TXnb(uint8_t * data, uint8_t size, SX12XX_Rad
   //   DBGLN("abort TX");
   //   return; // we were already TXing so abort. this should never happen!!!
   // }
-  SetMode(SX127x_OPMODE_STANDBY);
+  SetMode(SX127x_OPMODE_STANDBY, SX12XX_Radio_All);
+
+  if (radioNumber == SX12XX_Radio_Default)
+  {
+      radioNumber = lastSuccessfulPacketRadio;
+  }
 
-  hal.TXenable();
-  hal.writeRegister(SX127X_REG_FIFO_ADDR_PTR, SX127X_FIFO_TX_BASE_ADDR_MAX);
-  hal.writeRegisterFIFO(data, size);
+  RFAMP.TXenable(radioNumber);
+  hal.writeRegister(SX127X_REG_FIFO_ADDR_PTR, SX127X_FIFO_TX_BASE_ADDR_MAX, radioNumber);
+  hal.writeRegister(SX127X_REG_FIFO, data, size, radioNumber);
 
-  SetMode(SX127x_OPMODE_TX);
+  SetMode(SX127x_OPMODE_TX, radioNumber);
 }
 
 ///////////////////////////////////RX Functions Non-Blocking///////////////////////////////////////////
 
-void ICACHE_RAM_ATTR SX127xDriver::RXnbISR()
+bool ICACHE_RAM_ATTR SX127xDriver::RXnbISR(SX12XX_Radio_Number_t radioNumber)
 {
-  hal.readRegisterFIFO(RXdataBuffer, PayloadLength);
+  uint8_t const FIFOaddr = hal.readRegister(SX127X_REG_FIFO_RX_CURRENT_ADDR, radioNumber);
+  hal.writeRegister(SX127X_REG_FIFO_ADDR_PTR, FIFOaddr, radioNumber);
```

Comment:
Added `FIFOaddr` which is how the sx1280 driver does it.

There were issues with reading old packets from the FIFO of the other unread radio after packet reception.  `SX127X_REG_FIFO_ADDR_PTR` below did not appear to be setting the ptr correctly for both radios.